modules/Components/src/client/ACASThingBrowser.coffee

```diff
@@ -163,15 +189,34 @@ class ACASThingBrowserCellController extends Backbone.View
 	render: =>
 		$(@el).empty()
 
-		value = @model.get(@configs.key)
-		if value instanceof Value
-			content = value.escape("value")
-			if value.get("lsType") == "dateValue"  && !@configs.formatter?
-				content = UtilityFunctions::convertMSToYMDDate(content)
-		else if value instanceof Label
-			content = value.escape("labelText")
+		if @model instanceof Backbone.Model
+
+			# The model could be a thing or just a backbone model with key value pairs
+			value = @model.get(@configs.key)
+			# Render Thing Value
+			if value instanceof Value
+				content = value.get("value")
+				# If it's a string then escape it
+				if typeof(content) == "string"
+					content = value.escape(content)
+			# Render Thing Label
+			else if value instanceof Label
+				content = value.escape("labelText")
+			# Render key value backbone model
+			else 
+				# This is not a Thing Value or Label so assume it's just a backbone model with key/value pairs
+				# If the thing model class was passed in then we can use it to determine if this is a date value we should parse
+				content = @model.get(@configs.key)
+				if typeof(content) == "string"
+					content = @model.escape(@configs.key)
 		else
-			content =  @model.escape(@configs.key)
+			# This is strictly a key/pair object
+			content = @model[@configs.key]
+
+		if @options.thingModelExample?
+			modelValue = this.options.thingModelExample.get(@configs.key)
+			if modelValue instanceof Value && modelValue.get("lsType") == "dateValue" && !@configs.formatter?
+				content = UtilityFunctions::convertMSToYMDDate(content)
```

Comment:
There is a lot of magic here, but this is all about determining the type of response (Thing, plain backbone model, or plain javascript key/pair object) and getting their keys.  One of the huge slowdowns of this entire thing was that we were creating a collection of thousands of LsThings which are really slow to instantiate. The default Browser now works using plain javascript objects in the rows (flat response),  but it's possible to pass lsthings of Backbone objects to this class as well (without search browser controller).